GitUI/GitUICommands.cs

```diff
@@ -1755,8 +1756,10 @@ private bool RunFileHistoryCommand(IReadOnlyList<string> args, bool showBlame)
             }
             else
             {
+                // NOTE: fileHistoryFileName must be quoted.
+
                 ShowModelessForm(owner: null, requiresValidWorkingDir: true, preEvent: null, postEvent: null,
-                                 () => new FormFileHistory(commands: this, fileHistoryFileName, revision, filterByRevision, showBlame));
+                                 () => new FormFileHistory(commands: this, fileHistoryFileName.QuoteNE(), revision, filterByRevision, showBlame));
```

Comment:
> fileHistoryFileName must be quoted.

... but not if has already been quoted (by whichever shell for whatever reason - I cannot reproduce it with neither value of `AppSettings.UseBrowseForFileHistory`)

```
                // NOTE: fileHistoryFileName must be quoted.
                if (!fileHistoryFileName.Contains('"'))
                {
                    fileHistoryFileName = fileHistoryFileName.QuoteNE();
                }
```

Reply:
This code is invoked from here:
https://github.com/gitextensions/gitextensions/blob/21345f29511f71fb1ca7cfd8ea128972b56863e4/GitUI/GitUICommands.cs#L1449-L1456

`fileHistoryFileName` comes from `arg[2]`. On a command prompt `foo` or `"foo"` are treated the same - `foo`. We don't supply `"\"foo\""` on a command line and; if a user does it explicitly - then it's user's doing.

To debug it, you'd need to start the app like this:
```json
    "FileHistory": {
      "commandName": "Project",
      "commandLineArgs": "filehistory \"\\\"GitExtensions.settings\\\"\" "
    }
```


I considered adding a method that would quote if no quotes present, however, it would require a deeper investigation, which I don't have time for.

Reply:
(I could not reproduce it before because the shell extension started the wrong version, although I re-registered the shell extension and restarted the TotalCommander.)
Using the command line, I can reproduce it now and have verified that this PR fixes the issue.